src/author/author.controller.ts

```diff
@@ -0,0 +1,29 @@
+import { Controller, Post, Body, BadRequestException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { Authors } from './author.entity';
+import { CreateAuthorDto, CreatedAuthorDto } from './dto/create';
+
+@Controller('author')
+export class AuthorController {
+
+  constructor(
+    @InjectRepository(Authors)
+    private readonly authorsRepository: Repository<Authors>,
+  ) { }
+
+  @Post()
+  async create(@Body() authorInput: CreateAuthorDto): Promise<CreatedAuthorDto | BadRequestException> {
+
+    return await this.authorsRepository
+    .save(authorInput)
```

Comment:
Ah, animal isso aqui. Como o CreateAuthorDto respeita a interface do Autor você conseguiu passar ele direto para o save, é isso? Mesmo calma ê, como que o orm vai saber sobre as configurações que você colocou anotada com @Entity? Tipo, como ele sabe o nome da tabela, coluna que representa chave primária etc?

Reply:
Pelo que vi ele não bate as interfaces, apesar de eu ter criado o tipo DTO. Se eu mandar um objeto que não tenha nenhuma propriedade, ele grava igual, à não ser que viole uma regra do banco de dados. Então se vc enviar um objeto com propriedades a mais, elas serão ignoradas, e se for do mesmo nome esperado da entidade ele popula a coluna da tabela.

O próprio TypeORM que cria a tabela com as regras passadas nas anotações das propriedades da classe que tem o `@Entity`, e a propriedade `id` tem a anotação `@PrimaryGeneratedColumn('uuid')` que já garante a unicidade e ser primary key

Reply:
Entendi. Ah, o tipo Repository é do TypeOrm. Ele sabe que o repositório é de autor então consegue pegar as informações que vem no dto e associar corretamente com as colunas do banco de dados.